src/OpenTelemetry/Trace/ActivitySourceAdapter.cs

```diff
@@ -0,0 +1,81 @@
+﻿// <copyright file="ActivitySourceAdapter.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Diagnostics;
+using OpenTelemetry.Trace.Export;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// This class encapsulates the logic for performing ActivitySource actions
+    /// on Activities that are created using default ActivitySource.
+    /// All activities created without using ActivitySource will have a
+    /// default ActivitySource assigned to them with their name as empty string.
+    /// This class is to be used by instrumentation adapters which converts/augments
+    /// activies created without ActivitySource, into something which closely
+    /// matches the one created using ActivitySource.
+    /// </summary>
+    public class ActivitySourceAdapter
+    {
+        private ActivitySampler activitySampler;
+        private ActivityProcessor activityProcessor;
+
+        internal ActivitySourceAdapter(ActivitySampler activitySampler, ActivityProcessor activityProcessor)
+        {
+            this.activitySampler = activitySampler;
+            this.activityProcessor = activityProcessor;
+        }
+
+        private ActivitySourceAdapter()
```

Comment:
nit: I don't think we need the second ctor, didn't see any code using it and having the `internal` one without a `public` ctor effectively stops anyone from outside the library from making instances which is why I'm assuming the `private` was added?

Reply:
I was trying to prevent anyone from instantiating it! Didnt realize that its already achieved without private empty c..tor.

Reply:
Ya if you have no ctors at all than the compiler will basically add the implied `public ctor() {}` but as soon as you drop a real one in there (internal ctor in our case), it won't do that.